### PR TITLE
test: integration test

### PR DESCRIPTION
**Pull request by pmakani**

### CODE REVIEW

File: google-cloud-vision/src/test/java/com/google/cloud/vision/it/ITSystemTest.java [hunk 1]
```diff
@@ -0,0 +1,915 @@
...
+    List<AnnotateImageResponse> responses = response.getResponsesList();
+    for (AnnotateImageResponse res : responses) {
+      for (FaceAnnotation annotation : res.getFaceAnnotationsList()) {
+        assertEquals(Likelihood.POSSIBLE, annotation.getAngerLikelihood());
+        assertEquals(Likelihood.POSSIBLE, annotation.getJoyLikelihood());
+        assertEquals(Likelihood.LIKELY, annotation.getSurpriseLikelihood());
+      }
+    }
+  }
+
```
Comment (fazunenko) [inline google-cloud-vision/src/test/java/com/google/cloud/vision/it/ITSystemTest.java:231]: Overall test looks very good to me. 
One thing could be improved:

There are many tests staring with very similar code:

```
    ImageSource imgSource =
        ImageSource.newBuilder().setGcsImageUri(SAMPLE_BUCKET + "text/screen.jpg").build();
    Image img = Image.newBuilder().setSource(imgSource).build();
    Feature feat = Feature.newBuilder().setType(Type.DOCUMENT_TEXT_DETECTION).build();
    AnnotateImageRequest request =
        AnnotateImageRequest.newBuilder().addFeatures(feat).setImage(img).build();
    BatchAnnotateImagesResponse response =
        imageAnnotatorClient.batchAnnotateImages(ImmutableList.<AnnotateImageRequest>of(request));
    List<AnnotateImageResponse> responses = response.getResponsesList();

```
Would it make sense to introduce a function accepting several parameters and returning `List<AnnotateImageResponse>`?
Most of tests will become two times shorter?

**Review by fazunenko — CHANGES_REQUESTED** [review]

File: google-cloud-vision/src/test/java/com/google/cloud/vision/it/ITSystemTest.java (hunk 1 above)
Comment (pmakani) [reply google-cloud-vision/src/test/java/com/google/cloud/vision/it/ITSystemTest.java:231]: I tried you suggestion and run test cases couple of time but some times it works and some times not so keeping as it it.

**Review by athakor — APPROVED** [review]
LGTM

File: google-cloud-vision/src/test/java/com/google/cloud/vision/it/ITSystemTest.java (hunk 1 above)
Comment (dmitry-fa) [reply google-cloud-vision/src/test/java/com/google/cloud/vision/it/ITSystemTest.java:231]: My approach doesn't suggest any changes in behavior. It's only about refactoring. So tests should work as previously.

**Comment by pmakani** [comment]
@dmitry-fa PTAL

**Comment by dmitry-fa** [comment]
Thanks for changing!  This is exactly what I asked for.
A couple of extra notices: 
1) `public void detectLandmarksUrlTest` could be also refactored. (you probably skipped it)
2) detectWebEntitiesIncludeGeoResultsTest / detectWebEntitiesIncludeGeoResultsGcsTest

Would you mind adding support for them.
I mean:

Update your helper function to support extra paramater
```
private static List<AnnotateImageResponse> 
        getResponsesList(String image, Type type, boolean isGcs, ImageContext imageContext) throws IOException {
     ...
    AnnotateImageRequest request = imageContext == null ?
         AnnotateImageRequest.newBuilder().addFeatures(feat).setImage(img).build() :
         AnnotateImageRequest.newBuilder().addFeatures(feat).setImageContext(imageContext).setImage(img).build();
    ...
}
```

and adding another simple function:
```
private static List<AnnotateImageResponse> getResponsesList(String image, Type type, boolean isGcs) throws IOException {
   return getResponsesList(image, type,  isGcs, null);
}
```

**Pull request closed without merging**